synapse/handlers/identity.py

```diff
@@ -354,7 +354,8 @@ async def send_threepid_validation(
             raise SynapseError(500, "An error was encountered when sending the email")
 
         token_expires = (
-            self.hs.clock.time_msec() + self.hs.config.email_validation_token_lifetime
+            self.hs.get_clock().time_msec()
+            + self.hs.config.email_validation_token_lifetime
```

Comment:
I'm not sure if we want to encourage using it directly or not, but `self.clock` should exist also (the [`BaseHandler` object adds it](https://github.com/matrix-org/synapse/blob/de5cafe980391ae6e2de1d38ac4e42dea182a304/synapse/handlers/_base.py#L43)).

Reply:
Should I change it, or not?

Reply:
It is probably a bit outside the scope of this PR.